app/utils/domain_resolver.py

```
@@ -97,50 +156,150 @@ def check_megakino_domain_validity(base_url: str, timeout: float | int = 15) ->
             logger.warning(
                 "Megakino domain probe returned {} for {}", resp.status_code, probe_url
             )
-            return False
-        return True
+            return None
+        if not _is_sitemap_payload(resp.text or ""):
+            logger.warning("Megakino sitemap probe returned non-XML for {}", probe_url)
+            return None
+        return _normalize_domain(resp.url) or _normalize_domain(base_url)
     except Exception as exc:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)